app/helpers/nav_helper.rb

```diff
@@ -6,35 +6,68 @@ def nav_items
     sample_period = Work::Period.new(community: current_community)
     sample_job = Work::Job.new(period: sample_period)
     sample_shift = Work::Shift.new(job: sample_job)
-    items = [
-      {
+    # Build up the main menu list
+    items = []
+    # check if there are any main menu items defined in Settings.
+    c_settings_topmenu = current_community.settings.top_menu_customizations
+    settings_topmenu = if c_settings_topmenu.nil?
+                         []
+                       else
+                         # create an array of arrays from the setting string.
+                         c_settings_topmenu.scan(%r{\[([\w\- ]+)\]\(([\w/\.:%&\?]+)\)})
```

Comment:
If you call `.to_h` on this you'll actually get an ordered hash of the form `{"Foo" => "https://...}`

That might make things a little cleaner below.